scripts/schemaGenerator.py

```diff
@@ -55,6 +58,36 @@ def get_options_list_for_enum(field):
     return options_list
 
 
+def is_dynamic_config_supported_field(field, dbConfig=None):
+    if dbConfig is None or "dynamicConfigSupported" not in dbConfig:
+        # Case where dbConfig doesn't contain dynamicConfigSupported field at all
+        return False
+    dynamicConfigSupported = dbConfig["dynamicConfigSupported"]
+    configKey = ""
+    if "configKey" in field:
+        configKey = field["configKey"]
+    isDynamicConfigSupportedField = configKey in dynamicConfigSupported
+    return isDynamicConfigSupportedField
+
+
+def generate_uiconfig_pattern(field, dbConfig=None) -> str:
+    field_supports_dynamic_config = is_dynamic_config_supported_field(field, dbConfig)
+    destName = ""
+    if "name" in dbConfig:
+        destName = dbConfig["name"].lower()
+    # TODO: remove this once all the destinations have been updated with dynamicConfigSupported field
+    if destName not in CLEANED_DESTINATIONS:
+        return generalize_regex_pattern(field)
+
+    if field_supports_dynamic_config:
```

Comment:
this supports destinations which has been reviewed & db-config got updated with dynamicConfigSupported field and that has not been reviewed too
```suggestion
    if field_supports_dynamic_config || "dynamicConfigSupported" not in dbConfig:
```

Reply:
### We want satisfy below cases:

 - Destination in DYNAMIC_CONFIG_SUPPORTED_DESTINATIONS
	- Field supports dynamic config
		- Regex is present in ui-config for the field
			- Use the regex mentioned in ui-config & includes dynamic config regex & env regex(if not already present).
		- Regex is not present in ui-config for the field
			- Use the regex `^(.{0,100})$` & includes dynamic config regex & env regex(if not already present).
	- Field does not support dynamic config
		- Regex is present in ui-config for the field
			- Use the regex mentioned in ui-config.
		- Regex is not present in ui-config for the field
			- Use the regex `^(.{0,100})$`.
- Destination not in DYNAMIC_CONFIG_SUPPORTED_DESTINATIONS 
	 - Regex is present in ui-config for the field
		 - Use the regex mentioned in ui-config & includes dynamic config regex & env regex(if not already present).
	 - Regex is not present in ui-config for the field
		 - Use the regex `^(.{0,100})$` & includes dynamic config regex & env regex(if not already present).

I don't think this logic works well for these cases